addons/hr_expense/models/account_move.py

```diff
@@ -25,3 +25,7 @@ def _payment_state_matters(self):
         if self.line_ids.expense_id:
             return True
         return super()._payment_state_matters()
+
+    def button_cancel(self):
+        super(AccountMove, self).button_cancel()
+        self.expense_sheet_id.expense_line_ids.refuse_expense(_('Payment Refused'))
```

Comment:
With current fix:
- If we cancel the expense bill, both expense and related sheet will be canceled
- If we cancel the expense payment it will not affect the expense, because the payment move has no `expense_sheet_id` associated to it, but the expense is associated to the `account.move.line`, so we need something like 
```
self.line_ids.expense_id.refuse_expense(_('Payment Refused'))
```